include/OpenSpaceToolkit/Mathematics/CurveFitting/Interpolator.hpp

```diff
@@ -16,46 +18,60 @@ namespace curvefitting
 
 using ostk::core::type::Real;
 using ostk::core::type::Size;
+using ostk::core::type::Shared;
 
 using ostk::mathematics::object::VectorXd;
 
-/// @brief                      Interpolator
+/// @brief Interpolator (abstract class)
 ///
-///                             In mathematics, an interpolator is a type of estimator allowing to construct new data
-///                             points based on a range of a discrete set of known data points.
+/// In mathematics, an interpolator is a type of estimator allowing to construct new data
+/// points based on a range of a discrete set of known data points.
 ///
-/// @ref                        https://en.wikipedia.org/wiki/Interpolator.
-
+/// @ref https://en.wikipedia.org/wiki/Interpolator.
 class Interpolator
 {
    public:
-    /// @brief              Default constructor
-    ///
-    Interpolator() = default;
+    enum class InterpolationType
```

Comment:
```suggestion
    enum class Type
```
Since it is namespaced with Interpolator we can just call it Type.